charts/shoot-core/components/charts/network-policies/templates/node-exporter-to-api-server.yaml

```diff
@@ -0,0 +1,21 @@
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  annotations:
+    gardener.cloud/description: |
+      Allows Egress from node-exporter to api-server
+  name: gardener.cloud--allow-from-node-exporter-to-api-server
+  namespace: kube-system
+  labels:
+    origin: gardener
+spec:
+  podSelector:
+    matchLabels:
+      component: node-exporter
+  policyTypes:
+  - Egress
+  egress:
+  - to:
+    - podSelector:
+        matchLabels:
+          role: apiserver
```

Comment:
Node exporter talks via the API server on it's public IP.  Use https://github.com/gardener/gardener/blob/4dab9070a5ece73b2a6ad1a585a036c4a410108b/charts/shoot-core/components/charts/network-policies/templates/allow-to-public-networks.yaml#L1-L21 instead and label this pod with `networking.gardener.cloud/to-public-networks: allowed`